contrib/mpi-proxy-split/two-phase-algo.cpp

```diff
@@ -230,8 +230,10 @@ TwoPhaseAlgo::preSuspendBarrier(const void *data)
     case INTENT:
       setCkptPending();
       if (getCurrState() == PHASE_1) {
-        // If in PHASE_1, wait for us to finish doing IN_CS
-        if (waitForNewStateAfter(PHASE_1) != IN_CS) {
+	st = waitForNewStateAfter(PHASE_1, 10 /* timeout ms*/);
```

Comment:
Since this is a rare occurrence, let's set the timeout to 100 ms.  I'm worried that a timeout of 10 ms might add an extra 2% or 3% runtime overhead in some cases.